tests/plugins/speculation-rules/speculation-rules-test.php

```diff
@@ -19,6 +19,11 @@ public function tear_down() {
 		parent::tear_down();
 	}
 
+	public function test_hooks() {
```

Comment:
Good idea. I don't think we are validating that hooks are always present for each of the plugins. Should this type of test be added everywhere, since I don't believe we can run tests against the output of `wp_head` to confirm that these are actually being printed without introducing side effects?

Reply:
@joemcgill I generally like to have this kind of test as it helps prevent accidents with hooks added or removed, but I wouldn't want to mandate it. IMO it's a good practice, but I still think this is something we can leave up to the typical freedom of choice of the respective plugin developers / PR authors and PR reviewers.

I only added this here since the hook check previously was baked into the generator test method, which IMO it shouldn't be since it's a different kind of test.